### DISCUSSION TITLE
devcontainer feature caching

### DISCUSSION THREAD

**Discussion by lukewiwa** [devcontainers/community#22] [Q&A] [answered]
I often rebuild containers that I am working in for various reasons. I enjoy abstracting away dev dependencies into features but it can take a while to rebuild when I have a lot of heavy features tacked onto the end of the docker build steps. I understand features are essentially extra dockerfile commands run on the end of whatever container we choose so caching is tricky given the constraints but I'd like to explore the solutions.

Previously I'd put these dev dependencies earlier in the dockerfile build steps so that it would automatically cache. Is there some solution similar to this we can use?

**Comment by Chuxel** [comment, marked as answer]
Yeah this is being tracked in https://github.com/devcontainers/cli/issues/291 - right now Features are not independently cached. So caching is working, but the Features layer is cached together.
